src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ExpressionMetadataProvider.cs

```diff
@@ -102,6 +104,12 @@ public static class ExpressionMetadataProvider
                 //    m => m.Color.Red (nested property access)
                 //    m => m.Widgets[0].Size (expression ending with property-access)
                 metadata = metadataProvider.GetMetadataForType(containerType).Properties[propertyName];
+                if (metadata == null)
+                {
+                    throw new InvalidOperationException(Resources.FormatModelExpression_InvalidProperty(
+                        nameof(ModelMetadata),
+                        containerType.Name + "." + propertyName));
```

Comment:
> Need something better than $"{type}.{property}". That isn't valid C# unless the property is static. 

What's your suggestion? This doesn't need to be compilable code, it needs to tell the user what to fix.

> It also won't mean much when the property doesn't exist at all or is a field. 

You won't get here in that case. Non-existent property would be a compilation error. A field would go into the `if` and get the metadata for the field's type.

It's possible that this behavior isn't exactly right today for fields. Let's discuss.

>Suggest passing the property and type names to the resource separately.

We generally don't put things that look like code into a resource because it's easy for it be localized wrongly.  Let's discuss in person what your ideas are for improving this error message.